### PR TITLE
Leaves - Dianna

### PR DESCRIPTION
**Pull request by dtingg**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The model handles data and business logic for the application.
Describe in your own words what the Controller is doing in Rails | The controller manages the user interface for the application.
Describe in your own words what the View is doing in Rails | The view handles the graphical interface and presentation for the application.
Describe an edge-case controller test you wrote | If the user tries to see a task using an invalid id, I make sure that my application responds with a redirect to the root path.
What is the purpose of using strong params? (i.e. the params method in the controller) | Strong params are required to increase the security of our application.  Without them, a hacker could insert potentially dangerous data into our database.
How are Rails migrations related to Rails models? | A model is used to add persistence to our application.  The migration provides the instructions to the database on how to store the data the model is describing.
Describe one area of Rails that are still unclear on | I could probably use more help using the Rails debugger in VS Code.

### CODE REVIEW

File: test/controllers/tasks_controller_test.rb [hunk 1]
```diff
@@ -2,125 +2,181 @@
...
   describe "toggle_complete" do
-    # Your tests go here
+    it "correctly changes a task from complete to not_complete" do
+      patch not_complete_task_path(task.id)
+      
+      updated_task = Task.find_by(id: task.id)
+      
+      assert_nil(updated_task.completed)
+    end    
+
```
Comment (dHelmgren) [inline test/controllers/tasks_controller_test.rb:170]: What happens if the ID doesn't exist?

**Comment by dHelmgren** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | yes
Answered comprehension questions | yes
Successfully handles: Index, Show | yes
Index & Show tests pass | yes
Successfully handles: New, Create | yes
New & Create tests pass | yes
Successfully handles: Edit, Update | yes
Tests for Edit & Update test valid & invalid task ids | yes
Successfully handles: Destroy, Task Complete | yes
Tests for Destroy & Task Complete include tests for valid and invalid task ids | No test for missing ID in task complete
Routes follow RESTful conventions | yes
Uses named routes (like `_path`) | yes
**Overall** | Good job on Task List! It functions as expected! You hit all of the learning goals: using partials, using strong params, and following Rails conventions.
